### PR TITLE
Feature 47/assign team leader

### PR DESCRIPTION
**Pull request by rhacp**
Changed teamLeader to Long(userId) so even if a user changes his details, the teamLead field will still point to the correct user. Added API to assign teamLeader. Added validation for team already having a teamLeader and validation for PM not setting him up as TL.

### CODE REVIEW

File: src/main/java/com/teamrocket/tms/services/user/UserServiceImpl.java [hunk 1]
```diff
@@ -122,4 +121,25 @@ public TeamDTO createTeam(Long userId, TeamDTO teamDTO) {
...
+        log.info("User {} : {} retrieved. From assignTeamLeader.", userId, user.getLastName());
+        userServiceValidation.validateUserRoleCanPerformAction(user, Role.PROJECTMANAGER);
+
+        User leader = userRepository.findById(leaderId)
+                .orElseThrow(() -> new UserNotFoundException("User with the id " + leaderId + " not found."));
+        log.info("User {} : {} retrieved. From assignTeamLeader.", userId, user.getLastName());
+
+        userServiceValidation.validateAreUsersEquals(user, leader);
+
+        leader.setRole(Role.TEAMLEADER);
```
Comment (GabriellaNuca2209) [inline src/main/java/com/teamrocket/tms/services/user/UserServiceImpl.java:138]: Maybe we can do something about the case that a junior/medior is assigned as a team leader?

File: src/main/java/com/teamrocket/tms/services/user/UserServiceImpl.java [hunk 2]
```diff
@@ -122,4 +121,25 @@ public TeamDTO createTeam(Long userId, TeamDTO teamDTO) {
...
     }
+
+    @Override
+    public TeamDTO assignTeamLeader(Long userId, Long teamId, Long leaderId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new UserNotFoundException("User with the id " + userId + " not found."));
+        log.info("User {} : {} retrieved. From assignTeamLeader.", userId, user.getLastName());
+        userServiceValidation.validateUserRoleCanPerformAction(user, Role.PROJECTMANAGER);
+
+        User leader = userRepository.findById(leaderId)
```
Comment (Daniel7Ionescu) [inline src/main/java/com/teamrocket/tms/services/user/UserServiceImpl.java:132]: Maybe we should use targetUser?

File: src/main/java/com/teamrocket/tms/services/user/UserServiceImpl.java (hunk 1 above)
Comment (Daniel7Ionescu) [inline src/main/java/com/teamrocket/tms/services/user/UserServiceImpl.java:138]: check team before assigning role to user

**Review by GabriellaNuca2209 — CHANGES_REQUESTED** [review]
Please consider validations before implementing role changes

**Review by Daniel7Ionescu — APPROVED** [review]
Looks good, should have also mentioned that you are commenting out the direct APIs

**Review by GabriellaNuca2209 — APPROVED** [review]

**Pull request merged**